main/src/library/File.flix

```diff
@@ -59,4 +165,196 @@ namespace File {
+                let strLength = String.length(str);
+                String.dropRight(strLength - count, str)
+        }
+
+
+    ///
+    /// Returns an iterator of the given `file`
+    ///
+    pub def readLinesIter(file: String): Iterator[Result[String, String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path);
+            import java.io.BufferedReader.readLine();
+
+            let javaFile = newFile(file);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath);
+            let line = ref readLine(reader);
+
+            let done = () -> Object.isNull(deref line);
+            let next = () -> {
+                let l = deref line;
+                line := readLine(reader);
+                Ok(l)
+            };
+            Iterator(done, next)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                let err = ref getMessage(ex);
+                let done = () -> Object.isNull(deref err);
+                let next = () -> {
+                    Err(deref err)
+                };
+                Iterator(done, next)
+        }
+
+
+    ///
+    /// Returns an array of all the bytes in the given file `file`.
+    ///
+    pub def readBytes(file: String): Result[Array[Int8], String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readAllBytes(##java.nio.file.Path);
+
+            let javaFile = newFile(file);
+            let javaPath = toPath(javaFile);
+            let bytes = readAllBytes(javaPath);
+
+            Ok(bytes)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+
+    ///
+    /// Returns an iterator of the bytes in the given `file` in chunks of size `chunkSize`.
+    ///
+    pub def readChunks(file: String, chunkSize: Int32): Iterator[Result[Array[Int8], String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readAllBytes(##java.nio.file.Path);
+
+            let javaFile = newFile(file);
+            let javaPath = toPath(javaFile);
+            let bytes = readAllBytes(javaPath);
+            let byteCount = Array.length(bytes);
+            let i = ref 0;
+
+            let done = () -> deref i >= byteCount;
+            let next = () -> {
+                let j = deref i;
+                if(j < byteCount) {
+                    let end = {
+                        if(j+chunkSize >= byteCount) byteCount-1
+                        else j+chunkSize-1
```

Comment:
I don't think the style guide contains this, but usually we have spaces around operators and space after `if`